### PR TITLE
Adds h5delete tool

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

**Comment by lrknox** [comment]
Needs MANIFEST entry - ubuntu test fails.

**Comment by derobins** [comment]
> Needs MANIFEST entry - ubuntu test fails.

Fixed!

**Review by qkoziol — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -560,6 +560,20 @@ New Features
...
     Tools:
     ------
+    - Added h5delete tool
+
+        Deleting HDF5 storage when using the VOL can be tricky when the VOL
+        does not create files. The h5delete tool is a simple wrapper around
+        the H5Fdelete() API call that uses the VOL specified in the
+        HDF5_VOL_CONNECTOR environment variable to delete a "file". If
+        the call to H5Fdelete() fails, the tool will attempt to use
+        the POSIX remove(3) call call to remove the file.
```
Comment (lrknox) [inline release_docs/RELEASE.txt]: I think the meaning would be clear with just "...use the POSIX remove(3) call to remove the file."

**Review by lrknox — APPROVED** [review]

File: release_docs/RELEASE.txt (hunk 1 above)
Comment (derobins) [reply release_docs/RELEASE.txt]: Typo - will fix.

Comment (derobins) [reply release_docs/RELEASE.txt]: Fixed

**Pull request merged**